gym-service/src/test/java/org/example/integration/steps/trainee/TraineeSteps.java

```diff
@@ -0,0 +1,161 @@
+package org.example.integration.steps.trainee;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import lombok.RequiredArgsConstructor;
+import org.json.JSONObject;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.example.entities.User;
+import org.example.integration.config.TestContext;
+import org.example.repositories.UserRepository;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.example.integration.util.Constants.AUTHORIZATION_HEADER;
+import static org.example.integration.util.Constants.CONTENT_TYPE_JSON;
+
+@RequiredArgsConstructor
+public class TraineeSteps {
+    private static final String TRAINEE_PROFILE_URL = "/trainee/profile";
+    private static final String CHANGE_PASSWORD_URL = "/trainee/profile/changePassword";
+    private static final String TOGGLE_STATUS_URL = "/trainee/profile/toggleStatus";
+
+
+    private final TestContext testContext;
+    private final MockMvc mockMvc;
+    private final UserRepository userRepository;
+    private MvcResult mvcResult;
+
+    @Given("the trainee is not authenticated with username {string}")
+    public void theTraineeIsNotAuthenticated(String username) {
+        testContext.setJwtToken(null);
+    }
+
+    @When("the trainee requests the trainee profile")
+    public void theTraineeRequestsTheTraineeProfile() throws Exception {
+        mvcResult = mockMvc.perform(MockMvcRequestBuilders.get(TRAINEE_PROFILE_URL)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .header(HttpHeaders.AUTHORIZATION, AUTHORIZATION_HEADER + testContext.getJwtToken())
+                        .contentType(CONTENT_TYPE_JSON))
+                .andReturn();
+    }
+
+    @When("the trainee requests to change the password with a valid new password {string}")
+    public void theTraineeRequestsToChangeThePasswordWithAValidNewPassword(String password) throws Exception {
+        performPasswordChangeRequest(password);
+    }
+
+    @When("the trainee requests to change the password with an invalid new password {string}")
```

Comment:
"when" here and above are absolutely the same, prefer to have generic method instead. scenario name already describes that password is invalid